CRM/Aoservicelisting/Form/ProviderApplicationForm.php

```diff
@@ -83,22 +81,47 @@ public function setDefaultValues() {
 
           $defaults[$field] = $organization[$field];
         }
-        // Set the first work address field to be that from the primary contact. 
-        $organizationAddreses = civicrm_api3('Address', 'get', ['contact_id' => $this->organizationId, 'options' => ['sort' => "is_primary DESC"]])['values'];
+        $phoneIds = $addressRowsMissing = [];
+        // Fill in the address fields starting with the primary address for the first row
+        $organizationAddreses = civicrm_api3('Address', 'get', ['contact_id' => $this->organizationId, 'options' => ['sort' => "is_primary DESC"], 'return' => ['street_addres', 'postal_code', 'city', PHONE_ID_CUSTOM_FIELD]])['values'];
         if (!empty($organizationAddreses)) {
           foreach ($organizationAddreses as $orgAddress) {
             $defaults['work_address[' . $addressRowCount . ']'] = $orgAddress['street_address'];
             $defaults['postal_code[' . $addressRowCount . ']'] = $orgAddress['postal_code'];
             $defaults['city[' . $addressRowCount . ']'] = $orgAddress['city'];
+            // If the address is linked to a phone use that phone id to fill in the phone field in this row.
+            if (!empty($orgAddress[PHONE_ID_CUSTOM_FIELD])) {
+              $phone = civicrm_api3('Phone', 'get', ['id' => $orgAddress[PHONE_ID_CUSTOM_FIELD], 'sequential' => 1])['values'];
+              if (!empty($phone)) {
+                $defaults['phone[' . $addressRowCount . ']'] = $phone[0]['phone'];
+                $phoneIds[] = $orgAddress[PHONE_ID_CUSTOM_FIELD];
+              }
+            }
+            else {
+              $addressRowsMissing[] = $addressRowCount;
+            }
             $addressRowCount++;
           }
         }
-        $phoneRowId = 2;
-        $phones = civicrm_api3('Phone', 'get', ['contact_id' => $this->organizationId, 'phone' => ['!=' => $primaryContactPhone['phone']], 'options' => ['sort' => "is_primary DESC"]])['values'];
-        if (!empty($phones)) {
-          foreach ($phones as $phone) {
-            $defaults['phone[' . $phoneRowId . ']'] = $phone['phone'];
-            $phoneRowId++;
+        // If we have address field rows missing phones lets just start by filling in with any phones from the service listing contact
```

Comment:
@Edzelopez I included this so its a bit of a fall back approach right so that for current SLs on edit something should prefill even if not 100% correct